fastparquet/dataframe.py

```diff
@@ -126,9 +125,9 @@ def cat(col):
             raise ValueError('If using an index, must give an index name')
         if str(t) == 'category':
             # https://github.com/dask/fastparquet/issues/576#issuecomment-805579337
-            temp = Categorical([], categories=cat(col), fastpath=True)
+            temp = Categorical.from_codes([], categories=cat(col))
             vals = np.zeros(size, dtype=temp.codes.dtype)
-            c = Categorical(vals, dtype=temp.dtype, fastpath=True)
+            c = Categorical.from_codes(vals, dtype=temp.dtype)
```

Comment:
Can I please have categorical proof (!) that this does not cause a copy and temporary memory spike?

Reply:
This calls exactly the same functions as long as you provide a NumPy array as codes and not a pandas ExtensionArray plus one cheap validation and one ``np.asarray`` call that is no-copy.

```
dtype = pd.CategoricalDtype(["a", "b", "c"])
codes = np.array([0, 1, 2], dtype=np.int8)
cats = dtype.categories
cat = Categorical(codes, dtype=dtype, fastpath=True)
assert np.shares_memory(cats._values, cat.categories._values)
assert np.shares_memory(codes, cat.codes)

cat = Categorical.from_codes(codes, dtype=dtype)
assert np.shares_memory(cats._values, cat.categories._values)
assert np.shares_memory(codes, cat.codes)
```

This should show that no copy is made.